orderbook/src/api/handler.rs

```diff
@@ -27,16 +27,69 @@ pub struct FeeRequestBody {
     sell_token: H160,
 }
 
+#[derive(PartialEq, Serialize, Deserialize, Debug)]
+#[serde(rename_all = "camelCase")]
+pub struct OrderPostError {
+    error_type: String,
+    description: String,
+}
+
+#[derive(PartialEq, Serialize, Deserialize, Debug)]
+#[serde(rename_all = "UPPERCASE")]
+pub struct UidResponse {
+    uid: OrderUid,
+}
+
 pub async fn add_order(
     orderbook: Arc<OrderBook>,
     order: OrderCreation,
 ) -> Result<impl warp::Reply, Infallible> {
     let (body, status_code) = match orderbook.add_order(order).await {
-        Ok(()) => ("ok", StatusCode::CREATED),
-        Err(AddOrderError::AlreadyExists) => ("already exists", StatusCode::BAD_REQUEST),
-        Err(AddOrderError::InvalidSignature) => ("invalid signature", StatusCode::BAD_REQUEST),
-        Err(AddOrderError::PastNonce) => ("nonce is in the past", StatusCode::BAD_REQUEST),
-        Err(AddOrderError::PastValidTo) => ("validTo is in the past", StatusCode::BAD_REQUEST),
+        Ok(()) => (
+            warp::reply::json(&UidResponse {
```

Comment:
nit: I think currently the spec says order placement should return a string and not a dictionary

Reply:
dang it: it says:
```
application/json
```
https://github.com/gnosis/gp-v2-services/pull/49/files#diff-5016bdc926b6c0f649f363739d09159a1db664aae19a08fa6b38b03406f040dcR16

Hence, I thought that spec says we should return a json. but you are right that the type of UID is just a String. I will change it, sure.

Reply:
In that case shouldn't it return a JSON encoded string, so `"0x..."`?